beamer/state_machine.py

```diff
@@ -205,3 +232,57 @@ def _handle_claim_withdrawn(event: ClaimWithdrawn, context: Context) -> bool:
 
     claim.withdraw()
     return True
+
+
+def _handle_request_resolved(event: RequestResolved, context: Context) -> bool:
+    request = context.requests.get(event.request_id)
+    if request is None:
+        return False
+
+    request.l1_resolution_filler = event.filler
+
+    return False
+
+
+def _handle_fill_hash_invalidated(_event: FillHashInvalidated, _context: Context) -> bool:
+    # TODO: implement once contract is finished
+    return False
+
+
+def _l1_resolution_criteria_fulfilled(claim: Claim, context: Context) -> bool:
+    l1_resolution_gas_cost = 1_000_000  # FIXME
+    l1_gas_price = context.web3_l1.eth.gas_price
+    l1_safety_factor = 1.25
+    limit = int(l1_resolution_gas_cost * l1_gas_price * l1_safety_factor)
+
+    if claim.claimer == context.address:
+        if claim._latest_claim_made.challenger_stake > limit:
+            return True
+    elif claim.challenger == context.address:
+        if claim._latest_claim_made.claimer_stake > limit:
+            return True
+
+    return False
+
+
+def _handle_initiate_l1_resolution(event: InitiateL1ResolutionEvent, context: Context) -> bool:
```

Comment:
the function will never remove the event, and there are potentially multiple events for the same request, which could trigger the assertion error

Reply:
Good point, this should return `True`